code/part-one/mining.js

```diff
@@ -23,24 +34,45 @@ class MineableTransaction {
   }
 }
 
+/**
+ * Almost identical to the non-mineable block. In fact, we'll extend it
+ * so we can reuse the calculateHash method.
+ */
 class MineableBlock extends Block {
-  /*
-    Initialize the block constructor
-
-    **Note**
-    The block should be initiated with
-      - a timestamp
-      - transactions
-      - the previous block's hash
-  */
+  /**
+   * Unlike the non-mineable block, when this one is initialized, we want the
+   * hash and nonce to not be set. This Block starts invalid, and will
+   * become valid after it is mined.
+   */
   constructor(transactions, previousHash) {
     super(transactions, previousHash);
     this.hash = '';
     this.nonce = null;
   }
 }
 
+/**
+ * The new mineable chain is a major update to our old Blockchain. We'll
+ * extend it so we can use some of its methods, but it's going to look
+ * very different when we're done.
+ */
 class MineableChain extends Blockchain {
+  /**
+   * In addition initializing a blocks array with a genesis block, this will
+   * store hard coded difficulty and reward properties. These are settings
+   * used by the mining method.
```

Comment:
should we make `difficulty` and `reward` arguments that can be passed into the `constructor` function instead of hardcoding values directly?

Reply:
I don't think it makes much functional difference. In terms of the semantics/feel . . . I think hard-coded probably makes more sense. Bitcoin modifies these values algorithmically, but their starting values would have been hard-coded. I don't think you would expect to implement one chain with a reward of 100, another of 200, etc. What's you thinking?

Reply:
I guess a different way of phrasing my question (from curiosity, rather than "we should do it another way!") would've been "would we create different blockchains w/ different difficulty/rewards?" If you think no, then let's just leave as is!